changelog-maker.js

```diff
@@ -1,32 +1,49 @@
 #!/usr/bin/env node
 
-const gitcmd        = 'git log --pretty=full --since="{{sincecmd}}" --until="{{untilcmd}}"'
-    , commitdatecmd = '$(git show -s --format=%cd `{{refcmd}}`)'
-    , untilcmd      = ''
-    , refcmd        = 'git rev-list --max-count=1 {{ref}}'
-    , defaultRef    = '--tags=v*.*.*'
-
-
 const spawn    = require('child_process').spawn
     , bl       = require('bl')
     , split2   = require('split2')
     , list     = require('list-stream')
     , after    = require('after')
+    , fs       = require('fs')
+    , path     = require('path')
     , ghauth   = require('ghauth')
     , ghissues = require('ghissues')
     , chalk    = require('chalk')
+    , pkgtoId  = require('pkg-to-id')
+    , commitStream = require('./commit-stream')
+
     , argv     = require('minimist')(process.argv.slice(2))
-    , tsml     = require('tsml')
 
-    , commitStream = require('./commit-stream')
+    , quiet    = argv.quiet || argv.q || false
+
+    , pkg      = require('./package.json')
+    , debug    = require('debug')(pkg.name)
+
+    , cwd      = process.cwd()
+    , pkgFile  = path.join(cwd, 'package.json')
+    , pkgData  = fs.existsSync(pkgFile) ? require(pkgFile) : {}
+    , pkgId    = pkgtoId(pkgData)
 
-    , ghUser        = argv._[0] || 'nodejs'
-    , ghProject     = argv._[1] || 'io.js'
+    , ghId     = {
+          user: argv._[0] || pkgId.user || 'nodejs'
+        , name: argv._[1] || pkgId.name || 'io.js'
+      }
     , authOptions   = {
           configName : 'changelog-maker'
         , scopes     : ['repo']
       }
 
+const gitcmd        = 'git log --pretty=full --since="{{sincecmd}}" --until="{{untilcmd}}"'
+    , commitdatecmd = '$(git show -s --format=%cd `{{refcmd}}`)'
+    , untilcmd      = ''
+    , refcmd        = argv.a || argv.all ? 'git rev-list --max-parents=0 HEAD' : 'git rev-list --max-count=1 {{ref}}'
+    , defaultRef    = '--tags=v*.*.* 2> /dev/null ' +
+    '|| git rev-list --max-count=1 --tags=*.*.* 2> /dev/null ' +
```

Comment:
I'd like these two lines indented by another 2 characters but am not going to hold up the PR just for that